server/controllers/EventController.js

```diff
@@ -16,6 +15,64 @@ const getEvents = async (req, res, next) => {
 
   return res.json(events)
 }
+/*
+* Returns an array of JSON objects of a given length (10 if undefined), filled firs by all events that still need submitting, with the rest
+* of the array being filled with the most recent events ordered by date.
+* */
+const getEventDashboardList = async (req, res, next) => {
+
+  // Check if an array length number has been defined and store that value, otherwise default the value to 10
+  let listLength = (req.query.listLength !== undefined) ? req.query.listLength : 10
+
+  // Find & store all events from the collection
+  let allEvents = await getEventsFromDb()
+
+  // Find & store all events that still need submitting
+  let noRecordEvents = allEvents.filter((event) => {
+    return !event.hasAttendanceRecords
+  })
+
+  // Find & store all events that have allready been submitted
+  let recordEvents = allEvents.filter((event) => {
+    return event.hasAttendanceRecords
+  })
+
+  let sortedRecordEvents = recordEvents.sort((a, b) => {
+    const dateA = new Date(a.date)
+    const dateB = new Date(b.date)
+    return (dateA > dateB) ? -1 : 1
+  })
```

Comment:
`.sort()` mutates the original `recordEvents` array, which is why code below works even though you never used `sortedRecordEvents`.

We can just do:

```
recordEvents.sort((a, b) => {
  const dateA = new Date(a.date)
  const dateB = new Date(b.date)
  return (dateA > dateB) ? -1 : 1
})
```